akka-persistence-shared/src/test/scala/akka/persistence/serialization/SerializerSpec.scala

```diff
@@ -108,13 +108,13 @@ class SnapshotSerializerPersistenceSpec extends AkkaSpec(customSerializers) {
           "6024a8328a45e90200007870616263"
 
       val bytes = decodeHex(oldSnapshot.toCharArray)
-      val deserialized = serializer.fromBinary(bytes, None).asInstanceOf[Snapshot]
-
-      val deserializedDataStr = new String(deserialized.data.asInstanceOf[Array[Byte]], UTF_8)
-      dataStr should ===(deserializedDataStr)
+      val cause = intercept[NotSerializableException] {
+        serializer.fromBinary(bytes, None).asInstanceOf[Snapshot]
+      }
+      cause.getMessage should startWith("Replaying events from akka 2.3.x")
     }
 
-    "be able to read snapshot created with akka 2.3.6 and Scala 2.11" in {
+    "throw error when reads snapshot created with akka 2.3.6 and Scala 2.11" in {
```

Comment:
Could you add a similar test that verifies that latest code can read a snapshot created with 2.4.17 (or the commit before this PR). Just to be sure that the changes (readInt/writeInt) didn't break anything.